### ISSUE TITLE
[LS] Goto definition not working for statements with record de-structure binding patterns

### ISSUE THREAD

**Issue by NipunaRanasinghe** [ballerina-platform/ballerina-lang#34383]
**Description:**
Consider the following code sample.

```ballerina
type Person record {
    int age = 10;
};

public function main() {
    var {age} = fo<CURSOR>o();
}

function foo() returns Person {
    Person r = {age: 10};
    return r;
}
```
goto definition is not working for `foo()` in the above sample. (Feature is working when we have a simple binding pattern i.e. `var person = fo<CURSOR>o();`) 

**Steps to reproduce:**

**Affected Versions:**

**OS, DB, other environment details and versions:**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

**Comment by IMS94** [comment]
Here goto def for both function `foo()` and `age` (in mapping binding pattern) doesn't work at line:
```
var {age} = foo();
```

**Comment by pubudu91** [comment]
@manuranga Is this a blocker for the GA?

**Comment by manuranga** [comment]
Not a blocker

**Comment by nadeeshaan** [comment]
@pubudu91 @NipunaRanasinghe 
Here, the issue is with the location information of the `BLangRecordVariableDef`. The range of the `var {age} = foo();` should be from letter `v` to `;` and currently the captured position is the position range of `{age}`.

**Comment by pubudu91** [comment]
Fixed with #34592. Hence closing this.
